pyspark/test/bigdl/test_utils.py

```diff
@@ -214,13 +215,51 @@ def modelTest(self,
                              rtol=rtol,
                              atol=atol)
 
+        # compare gradient input
+        sess = K.get_session()
+        feed_dict = {K.learning_phase(): 1}
+        if isinstance(input_data, list):  # for merge layers
+            for i in range(0, len(input_data)):
+                feed_dict[keras_model.input[i]] = input_data[i]
+        else:
+            feed_dict[keras_model.input] = input_data
+        keras_grad_input = sess.run(K.gradients(keras_model.output * keras_output, keras_model.input),
+                                    feed_dict=feed_dict)
+        bigdl_grad_input = bigdl_model.backward(input_data, bigdl_output2)
+        if isinstance(bigdl_grad_input, list):  # for merge layers
```

Comment:
Extract this into two parts. one is for getting the grad, the other is for comparison.

Reply:
You mean we use two functions separately for doing this? But when comparing the forward output, we are not doing in two parts.